packages/react-component-library/src/components/DatePicker/CalendarTable.tsx

```diff
@@ -0,0 +1,148 @@
+import { IconChevronLeft, IconChevronRight } from '@royalnavy/icon-library'
+import { setMonth as changeMonth, setYear as changeYear } from 'date-fns/fp'
+import React, { useState, Dispatch, SetStateAction } from 'react'
+
+import { BUTTON_VARIANT } from '../Button'
+import { CALENDAR_TABLE_VIEW, type CalendarTableView } from './constants'
+import { DATEPICKER_ACTION, type DatePickerAction } from './types'
+import {
+  StyledButtonArrows,
+  StyledCalendarContainer,
+  StyledCalendarTable,
+  StyledCalendarTiles,
+  StyledNavigation,
+  StyledYearsTitle,
+} from './partials/StyledCalendarNavigation'
+
+interface CalendarTableProps {
+  calendarTable: CalendarTableView
+  dispatch: Dispatch<DatePickerAction>
+  month: Date
+  setCalendarTable: Dispatch<SetStateAction<CalendarTableView>>
+}
+
+const MONTHS = [
+  'JAN',
+  'FEB',
+  'MAR',
+  'APR',
+  'MAY',
+  'JUN',
+  'JUL',
+  'AUG',
+  'SEP',
+  'OCT',
+  'NOV',
+  'DEC',
+]
+
+const getDecade = (year: number) =>
+  Array.from({ length: 12 }, (_, i) => Math.floor(year / 10) * 10 + i - 1)
+
+const MonthsView = ({
+  month,
+  dispatch,
+  setCalendarTable,
+}: Pick<CalendarTableProps, 'month' | 'dispatch' | 'setCalendarTable'>) => {
+  const currentMonth = new Date().getMonth()
+  const selectedMonth = month.getMonth()
+
+  return (
+    <StyledCalendarTable $isMonths>
+      {MONTHS.map((label, index) => (
+        <StyledCalendarTiles
+          key={label}
+          aria-label={`Select ${label}`}
+          $isCurrent={index === currentMonth}
+          $isSelected={index === selectedMonth}
+          onClick={() => {
+            dispatch({
+              type: DATEPICKER_ACTION.UPDATE,
+              data: { currentMonth: changeMonth(index, month) },
+            })
+            setCalendarTable(CALENDAR_TABLE_VIEW.HIDE)
+          }}
+        >
+          {label.toUpperCase()}
```

Comment:
I don't think `toUpperCase` is required. Not really important.

Reply:
Hmm yeah I wondered if it was something to do with military date format.